CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityPlugin.cpp

```diff
@@ -452,7 +452,11 @@ double FocalPointPlasticityPlugin::tryAddingNewJunction(const Point3D &pt,const
 	// Randomly shuffle neighbor selection
 	std::vector<unsigned int> randNeighborIndex = std::vector<unsigned int>(maxNeighborIndexLocal + 1, 0);
 	for (unsigned int nIdx = 0; nIdx <= maxNeighborIndexLocal; ++nIdx) randNeighborIndex[nIdx] = nIdx;
-	std::random_shuffle(randNeighborIndex.begin(), randNeighborIndex.end());
+
+    // in a regular run we do randomly select a neighbor with which we make a new link. In test run or test data generate run we do not perform this randomization
+    if (!potts->get_test_output_generate_flag() && !potts->get_test_run_flag()) {
```

Comment:
If we were to record the seed and reuse it during testing, wouldn't this be unnecessary?

Reply:
Possibly. We can make separate PR testing this approach. For now I decided to play it safely. I would suggest making  an issue to track it

Reply:
Works for me.